packages/smithy-http/src/smithy_http/serializers.py

```diff
@@ -83,22 +85,33 @@ def begin_struct(self, schema: Schema) -> Iterator[ShapeSerializer]:
         if self._endpoint_trait is not None:
             host_prefix = self._endpoint_trait.host_prefix
 
+        content_type = self._payload_codec.media_type
+
         if (payload_member := self._get_payload_member(schema)) is not None:
             if payload_member.shape_type in (ShapeType.BLOB, ShapeType.STRING):
+                content_type = (
+                    "application/octet-stream"
+                    if payload_member.shape_type is ShapeType.BLOB
+                    else "text/plain"
+                )
                 payload_serializer = RawPayloadSerializer()
                 binding_serializer = HTTPRequestBindingSerializer(
                     payload_serializer, self._http_trait.path, host_prefix
                 )
                 yield binding_serializer
                 payload = payload_serializer.payload
             else:
+                if (media_type := payload_member.get_trait(MediaTypeTrait)) is not None:
+                    content_type = media_type.value
                 payload = BytesIO()
                 payload_serializer = self._payload_codec.create_serializer(payload)
                 binding_serializer = HTTPRequestBindingSerializer(
                     payload_serializer, self._http_trait.path, host_prefix
                 )
                 yield binding_serializer
         else:
+            if self._get_eventstreaming_member(schema) is not None:
```

Comment:
Might be handy (and more efficient) if we can embed this kind of information in the schemas themselves when we code-generate them, instead of having to do checks like this.

Reply:
`APIOperation.input_stream_member`